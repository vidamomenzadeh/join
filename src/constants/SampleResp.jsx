export const SAMPLE_RESP = {"1" :{
    "id": "1",
    "title": "Stolen Bike",
    "description": "Stolen Bike des.",
    "address": "Berlin",
    "occurred_at": 1554543180609,
    "updated_at": 1554543180609,
    "source": {
        "name": null,
        "html_url": null,
        "api_url": ""
    },
    "media": {
        "image_url": null,
        "image_url_thumb": null
    },
    "location_type": null,
    "location_description": null,
    "type": "Theft",
    "type_properties": null
}};

export const SAMPLE_RESP_2 = {"incidents":[{
        "id": "1",
        "title": "Stolen Bike",
        "description": "Stolen Bike des.",
        "address": "Berlin",
        "occurred_at": 1554543180609,
        "updated_at": 1554543180609,
        "source": {
            "name": null,
            "html_url": null,
            "api_url": ""
        },
        "media": {
            "image_url": null,
            "image_url_thumb": null
        },
        "location_type": null,
        "location_description": null,
        "type": "Theft",
        "type_properties": null
    }]};
