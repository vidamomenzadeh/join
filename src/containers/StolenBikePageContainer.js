import React from "react";
import { Component } from 'react';
import { connect } from 'react-redux';
import {getStolenBikesData} from '../actions/stolenBike';
import {withRouter} from "react-router";
import LayoutCmp from '@components/LayoutCmp';
import {getStolenBikes, getTotalBikes, getIsloading, getErrors} from "@reducers/stolenBikes";
import StolenBikeList from "@components/stolenBike/StolenBikeList";
import {WrappedBikeFilterForm} from "@components/stolenBike/FilterBike";

class StolenBikePageContainer extends Component {

    constructor(props){
        super(props);
        this.state = {
            filter : {
                page : 1
            }
        };

        this.applyStolenBikeFilter = this.applyStolenBikeFilter.bind(this);
        this.fetchBikesOnPage = this.fetchBikesOnPage.bind(this);
    }

    componentDidMount() {
        this.props.getStolenBikesData();
    }

    /*
    * Apply Bike Search Filtered
    * */
    applyStolenBikeFilter(filter){
        this.setState({
            filter : filter
        });
        this.props.getStolenBikesData(filter);
    }

    fetchBikesOnPage(page){
        this.setState({filter : {...this.state.filter, ...{page:page}}},()=>{
            this.props.getStolenBikesData(this.state.filter);
        });
    }

    render() {
        const {bikes, total, loading, errors} = this.props;

        return(
            <LayoutCmp  showFooter={true}
                        showHeader={true}>
                <div className="app">
                    <div className={"search-title-page"}>
                        Police Department of Berlin
                        <div className={"search-sub-title"}>Stolen bykes</div>
                    </div>
                    <WrappedBikeFilterForm applyStolenBikeFilter={this.applyStolenBikeFilter}/>
                    {errors && <div>Something went wrong...</div>}
                    {loading ? <div>Loading...</div>
                             : total == 0 ? <div>No Results...</div>
                                          :(<React.Fragment>
                                                <div className={"show-total-count"}>Total {total}</div>
                                            <StolenBikeList bikes={bikes}
                                            total={total}
                                            currentPage={this.state.filter.page}
                                            fetchBikesOnPage={this.fetchBikesOnPage}/>
                                           </React.Fragment>)

                    }
                </div>
            </LayoutCmp>
        )
    }
}

function mapStateToProps(state) {
    return {
        bikes : getStolenBikes(state),
        total : getTotalBikes(state),
        loading : getIsloading(state),
        errors : getErrors(state)
    }
}

export default withRouter(connect(mapStateToProps, {
    getStolenBikesData : getStolenBikesData
})(StolenBikePageContainer));
