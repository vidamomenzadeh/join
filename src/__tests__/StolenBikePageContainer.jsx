import React from "react";
import { shallow, mount } from 'enzyme';
import StolenBikePageContainer from '../containers/StolenBikePageContainer';
import {SAMPLE_RESP} from "@constants/SampleResp";
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });


describe('Snapshot test Container', () => {
    it('renders correctly', () => {
        const mapWrapper = shallow(<StolenBikePageContainer />);
        expect(mapWrapper).toMatchSnapshot();
    });

    it('should render correctly with bikes', () => {
        const bikes = SAMPLE_RESP;

        const mapWrapper = shallow(<StolenBikePageContainer bikes={bikes} />);
        expect(mapWrapper).toMatchSnapshot();
    });
});

