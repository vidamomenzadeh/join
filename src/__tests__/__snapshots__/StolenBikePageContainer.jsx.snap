// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Container renders correctly 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`Snapshot test Container should render correctly with markers 1`] = `
<Route>
  <Component />
</Route>
`;
